### PR TITLE
Initial Pass at Interface for MessageBus Abstraction

### PR DESCRIPTION
**Pull request by rsdmike**
Checkout interface.go for the heart of the interface.

### CODE REVIEW

**Comment by lenny-intel** [comment]
@jpwhitemn , corect. The user will transfer the configuration data from the services config from consul into the MessageBusConfig. Used this same pattern for the Registry.

File: interface.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+// limitations under the License.
+//
+
+package messaging
+
+// MessageClient is the messaging interface for publisher-subsriber pattern
+type MessageClient interface {
+	// Connect to messaging host specified in MessageBusconfig
+	// returns error if not able to connect
+	Connect(msgBusConfig *MessageBusConfig) error
```
Comment (lenny-intel) [inline interface.go]: remove msgBusConfig *MessageBusConfig. This will be passed to the factory.NewMessageClient

**Comment by andresrinivasan** [comment]
Late to the party, I know.

Given Go support for closure and the compiler/optimizer internally represents structures as pointers to minimize copying, I've grown to appreciate the esthetics and simplicity of always working with a struct. By extension, I suggest we return struct rather than returning *struct as well as never create side effects in function/method parameters.

**Comment by lenny-intel** [comment]
Closing as this was a preview and still WIP

**Pull request closed without merging**